### PR TITLE
More ssz objects pt 1

### PR DESCRIPTION
**Pull request by djrtwo**
addresses most of  #286 

* proof of custody data ssz object
* validator delta block ssz object
* _note_: didn't address the custody bit

### CODE REVIEW

**Comment by djrtwo** [comment]
wasn't quite sure how I wanted to do the custody bit. putting this up for review and will do the custody bit in another PR

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -377,6 +379,16 @@ Unless otherwise indicated, code appearing in `this style` is to be interpreted
 }
 ```
 
+#### `ProofOfPossessionData`
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: I would simply reuse `DepositInput `with `proof_of_possession` set to `[0, 0]`.

File: specs/core/0_beacon-chain.md [hunk 2]
```diff
@@ -967,17 +990,19 @@ def get_effective_balance(validator: ValidatorRecord) -> int:
 
 ```python
 def get_new_validator_registry_delta_chain_tip(current_validator_registry_delta_chain_tip: Hash32,
-                                               index: int,
+                                               validator_index: int,
                                                pubkey: int,
                                                flag: int) -> Hash32:
     """
-    Compute the next hash in the validator registry delta hash chain.
+    Compute the next root in the validator registry delta hash chain.
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: delta hash chain => delta root chain

File: specs/core/0_beacon-chain.md [hunk 3]
```diff
@@ -967,17 +990,19 @@ def get_effective_balance(validator: ValidatorRecord) -> int:
...
     """
-    return hash(
-        current_validator_registry_delta_chain_tip +
-        bytes1(flag) +
-        bytes3(index) +
-        bytes48(pubkey)
+    return hash_tree_root(
+        ValidatorRegistryDeltaBlock(
+            current_validator_registry_delta_chain_tip,
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: `validator_registry_delta_chain_tip= current_validator_registry_delta_chain_tip,`

File: specs/core/0_beacon-chain.md (hunk 1 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: While we're at it, maybe define the constant `EMPTY_SIGNATURE` to be `[0, 0]`. It's used three times.

File: specs/core/0_beacon-chain.md [hunk 4]
```diff
@@ -967,17 +990,19 @@ def get_effective_balance(validator: ValidatorRecord) -> int:
 
 ```python
 def get_new_validator_registry_delta_chain_tip(current_validator_registry_delta_chain_tip: Hash32,
-                                               index: int,
+                                               validator_index: int,
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: Also, need to update the keyword argument where that this function is being called.
(in `def activate_validator()` and in `def exit_validator()`)

File: specs/core/0_beacon-chain.md [hunk 5]
```diff
@@ -34,6 +34,7 @@
                 - [`Deposit`](#deposit)
                 - [`DepositData`](#depositdata)
                 - [`DepositInput`](#depositinput)
+                - [`ProofOfPossessionData`](#proofofpossessiondata)
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: That line should go

Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: agh!

**Pull request merged**